src/python/pants/goal/run_tracker.py

```diff
@@ -188,3 +190,7 @@ def retrieve_logs(self) -> List[str]:
             logger.warning("Error retrieving per-run logs from RunTracker.", exc_info=e)
 
         return output
+
+    @property
+    def counter_names(self) -> tuple[str, ...]:
```

Comment:
@tdyas I'm being inconsistent between "metric" vs. "counter". Should this property be called `metric_names` instead?

Reply:
The counter metrics are exposed as the `counters` property on workunits, so calling this `counter_names` is actually consistent.

Reply:
(Histograms for observation metrics are exposed via the `get_observation_histograms` function on some context FFI object. The Buildsense plugin has to "pull" the observation metrics if it wants to.)